File: Examples/intel_examples/examples_core_f/vmlf/source/errstatus.f
!===============================================================================
! Copyright 2001-2020 Intel Corporation.
!
! This software and the related documents are Intel copyrighted  materials,  and
! your use of  them is  governed by the  express license  under which  they were
! provided to you (License).  Unless the License provides otherwise, you may not
! use, modify, copy, publish, distribute,  disclose or transmit this software or
! the related documents without Intel's prior written permission.
!
! This software and the related documents  are provided as  is,  with no express
! or implied  warranties,  other  than those  that are  expressly stated  in the
! License.
!===============================================================================

!  Content:
!    vmlSetErrStatus/vmlGetErrStatus/vmlClearErrStatus  Example Program Text
!*******************************************************************************

      program MKL_VML_TEST

      include "mkl_vml.f90"

      integer errst
      integer(kind=4) terrst

      print *,"vmlSetErrStatus/vmlGetErrStatus/vmlClearErrStatus",      &
     &        " example program"
      print *,""
      print *,""

      errst=VMLGETERRSTATUS()

      print 10,"Initial value of vmlErrStatus: ",errst,"h"
      call PrintTextVmlErrStatus(errst)

      errst=VML_STATUS_BADMEM
      terrst=VMLSETERRSTATUS(errst)
      errst=VMLGETERRSTATUS()
      print 10,"Value of vmlErrStatus after using vmlSetErrStatus: ",   &
     &         errst,"h"
      call PrintTextVmlErrStatus(errst)

      terrst=VMLCLEARERRSTATUS()
      errst=VMLGETERRSTATUS()
      print 10,"Value of vmlErrStatus after using vmlClearErrStatus: ", &
     &         errst,"h"
      call PrintTextVmlErrStatus(errst)

10    format(A,Z16,A)

      end

      subroutine PrintTextVmlErrStatus(errst)

      include "mkl_vml.f90"

      integer errst

      if(errst.eq.VML_STATUS_OK       ) print *,"VML_STATUS_OK       "
      if(errst.eq.VML_STATUS_BADSIZE  ) print *,"VML_STATUS_BADSIZE  "
      if(errst.eq.VML_STATUS_BADMEM   ) print *,"VML_STATUS_BADMEM   "
      if(errst.eq.VML_STATUS_ERRDOM   ) print *,"VML_STATUS_ERRDOM   "
      if(errst.eq.VML_STATUS_SING     ) print *,"VML_STATUS_SING     "
      if(errst.eq.VML_STATUS_OVERFLOW ) print *,"VML_STATUS_OVERFLOW "
      if(errst.eq.VML_STATUS_UNDERFLOW) print *,"VML_STATUS_UNDERFLOW"

      end

File: Examples/intel_examples/examples_core_c/vmlc/source/errstatus.c
/*******************************************************************************
* Copyright 2001-2020 Intel Corporation.
*
* This software and the related documents are Intel copyrighted  materials,  and
* your use of  them is  governed by the  express license  under which  they were
* provided to you (License).  Unless the License provides otherwise, you may not
* use, modify, copy, publish, distribute,  disclose or transmit this software or
* the related documents without Intel's prior written permission.
*
* This software and the related documents  are provided as  is,  with no express
* or implied  warranties,  other  than those  that are  expressly stated  in the
* License.
*******************************************************************************/

/*
!  Content:
!    vmlSetErrStatus/vmlGetErrStatus/vmlClearErrStatus example program  Example
!    Program Text
!******************************************************************************/

#include <stdio.h>
#include "mkl_vml.h"

void PrintTextVmlErrStatus(int );

int main()
{
  MKL_INT errst;

  printf("vmlSetErrStatus/vmlGetErrStatus/vmlClearErrStatus example program\n\n");

  errst=(MKL_INT)vmlGetErrStatus();

  printf("Initial value of vmlErrStatus: ");
  PrintTextVmlErrStatus((int)errst);
  printf(" (0x%x)\n",(int)errst);

  errst=(MKL_INT)(VML_STATUS_BADMEM);
  vmlSetErrStatus(errst);
  errst=(int)vmlGetErrStatus();
  printf("Value of vmlErrStatus after using vmlSetErrStatus: ");
  PrintTextVmlErrStatus((int)errst);
  printf(" (0x%x)\n",(int)errst);

  vmlClearErrStatus();
  errst=(MKL_INT)vmlGetErrStatus();
  printf("Value of vmlErrStatus after using vmlClearErrStatus: ");
  PrintTextVmlErrStatus((int)errst);
  printf(" (0x%x)\n",(int)errst);

  return 0;
}

void PrintTextVmlErrStatus(int errst)
{
  switch(errst) {
    case VML_STATUS_OK: {
                printf("VML_STATUS_OK");
                break;
    }
    case VML_STATUS_BADSIZE: {
                printf("VML_STATUS_BADSIZE");
                break;
    }
    case VML_STATUS_BADMEM: {
                printf("VML_STATUS_BADMEM");
                break;
    }
    case VML_STATUS_ERRDOM: {
                printf("VML_STATUS_ERRDOM");
                break;
    }
    case VML_STATUS_SING: {
                printf("VML_STATUS_SING");
                break;
    }
    case VML_STATUS_OVERFLOW: {
                printf("VML_STATUS_OVERFLOW");
                break;
    }
    case VML_STATUS_UNDERFLOW: {
                printf("VML_STATUS_UNDERFLOW");
                break;
    }
  }
}
